math/program/src/uint.rs

```diff
@@ -0,0 +1,45 @@
+//! Large uint types
+
+// required for clippy
+#![allow(clippy::assign_op_pattern)]
+#![allow(clippy::ptr_offset_with_cast)]
+#![allow(clippy::unknown_clippy_lints)]
+#![allow(clippy::manual_range_contains)]
+
+use uint::construct_uint;
+
+construct_uint! {
+    pub struct U256(4);
+}
+construct_uint! {
+    pub struct U192(3);
+}
+
+impl U256 {
+    /// Returns self to the power of b
+    pub fn checked_u8_power(&self, b: u8) -> Option<U256> {
```

Comment:
This feels too inefficient to add to the library, binary decomposition is always faster right?

Reply:
This is being used by the stable curve implementation, and I didn't want to change it -- I'll move it back to swap

Reply:
Done in be1c771